spec/services/hyrax/custom_queries/navigators/collection_members_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+RSpec.describe Hyrax::CustomQueries::Navigators::CollectionMembers, valkyrie_adapter: :test_adapter do
+  subject(:query_handler) { described_class.new(query_service: query_service) }
+  let(:adapter)           { Hyrax.metadata_adapter }
+  let(:persister)         { adapter.persister }
+  let(:query_service)     { adapter.query_service }
+
+  describe '#find_collections_for' do
+    let(:work) { FactoryBot.valkyrie_create(:hyrax_work) }
+
+    it 'is empty with no collections' do
+      expect(query_handler.find_collections_for(resource: work))
+        .to be_empty
+    end
+
+    context 'when it is a member of collections' do
+      let(:work) { FactoryBot.valkyrie_create(:hyrax_work, :as_member_of_multiple_collections) }
+
+      it 'finds collections' do
+        expect(query_handler.find_collections_for(resource: work).map(&:id))
```

Comment:
Do we have an assertion around `:as_member_of_multiple_collections` functioning? I can see that if the factory has an empty array for collection membership, then this spec would succeed.

Reply:
fair point. i could add a separate spec for `expect(query_handler.find_collections_for(resource: work)).not_to be_empty`

other suggestions?

Reply:
One pattern that I had tried to work through was having a factory spec area, where we test the behavior of factories. Some of them can be rather gnarly. I don't know if that is part of the code-base. That also seems like a much larger exercise.

I believe a separate spec (or even assertion within spec) and a comment saying something along the lines of "Verifying working factory" (hmm... Maybe we need an `OSHA` gem?)